File: downloadablebookandexercisecode/1256/code_files/Chapter_03_Complexity_Analysis/CSES_missing_number.cpp
// https://cses.fi/problemset/task/1083

#include <iostream>

using namespace std;

int main() {
    long n; // b/c the n*(n+1) can overflow
    cin >> n;

    long sum = 0;
    for (int i = 0; i < n - 1; ++i) {
        int a;
        cin >> a;
        sum += a;
    }

    int missing_a = (n + 1) * n / 2 - sum;
    cout << missing_a << endl;
    return 0;
}

File: downloadablebookandexercisecode/1256/code_files/Chapter_03_Complexity_Analysis/CSES_missing_number.py
#  https://cses.fi/problemset/task/1083

n = int(input())

in_list = map(int, input().strip().split())

sum_list = sum(in_list)

sum_n = (n + 1) * n / 2

num = sum_n - sum_list

print(f'{int(num)}')
